### PR TITLE
IVR: move totalCanvases into context

### PR DESCRIPTION
**Pull request by davidpmccormick**
Part of #12986 

## What does this change?

Adds `totalCanvases` to the refactored `ItemViewerContext`, derived once in `IIIFViewer` alongside `currentCanvas`. `ViewerBottomBar` reads it instead of working the count out itself.

- `ViewerBottomBar` chose `totalCanvases` between two branches on `hasCompleteStructure`, but both arms compared `Object.keys(canvasIndexById).length` against `canvases.length` — they could only ever agree. That whole branch is deleted; it's just the canvas count.
- Its prev/next logic only ever compared the 1-based `canvas` number against that count. It never looked a canvas up by identity, so a reordering structure can't affect it.

The plan for this step also listed `isFirstCanvas`, `isLastCanvas`, `canNavigateNext` and `canNavigatePrevious` for the context, following the RFC. On checking the actual consumers, none of them meet the RFC's own criterion of "calculated in 2+ components":

`totalCanvases` does meet the bar: `canvases.length` is currently recomputed in ViewerTopBar, IIIFSearchWithin, Paginators, VirtualizedImageViewer, GridViewer and PaginatedItemViewer. This PR only converts `ViewerBottomBar`; the rest follow in step 3 (`hasMultipleCanvases`, three sites) and step 8 (the "page X of Y" label, four sites), which both need this value.

### Also here

`ViewerBottomBar`'s context import moves from the barrel to `ItemViewerContext/refactored`, matching what step 1 did to the other refactored components. It's only ever rendered under the refactored provider, so it doesn't need the flag-aware hook.

## How to test

- Tests: from `content/webapp`, `yarn jest views/pages/works/work/IIIFViewer test/fixtures/iiif hooks/useIIIFProbeService views/pages/works/work/work.helpers` — 181 passing, unchanged from step 1
- Full `yarn test:content` is 608 passing
- `yarn tsc` clean from the root

## Have we considered potential risks?

No biggies

### CODE REVIEW

File: content/webapp/views/pages/works/work/work.helpers.test.tsx [hunk 1]
```diff
@@ -114,3 +114,217 @@ describe('getCurrentCanvas', () => {
...
+      ).toMatchObject({
+        canNavigatePrevious: false,
+        canNavigateNext: false,
+      });
+    });
+  });
+
+  // Both invalid canvas numbers are reachable via the ?canvas= query param, so
+  // they need to degrade sensibly rather than offer navigation off either end.
+  describe('out-of-range canvas numbers', () => {
```
Comment (davidpmccormick) [inline content/webapp/views/pages/works/work/work.helpers.test.tsx]: Not sure if we want to leave either of the prev/next buttons functional if clicking them would take you to another out-of-range canvas? Would it make more sense for them both to be disabled? Don't know if it's worth changing the behaviour here, or marking as something to change later (or ignoring because it's an edge case and getting on with our lives)

Comment (rcantin-w) [reply content/webapp/views/pages/works/work/work.helpers.test.tsx]: We don't with our other pagination component (e.g. https://wellcomecollection.org/search/works?query=blue&page=300), but it could be an interesting one to tweak at some point. Feels quite niche though?

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/ViewerBottomBar.tsx [hunk 2]
```diff
@@ -135,26 +135,20 @@ const ViewerBottomBar: FunctionComponent = () => {
...
   } = useItemViewerContext();
 
   const { canvases } = { ...transformedManifest };
   const { canvas } = query;
 
-  const hasCompleteStructure =
-    Object.keys(canvasIndexById).length === canvases?.length;
-  const totalCanvases = hasCompleteStructure
-    ? Object.keys(canvasIndexById).length
-    : canvases?.length || 0;
```
Comment (rcantin-w) [inline content/webapp/views/pages/works/work/IIIFViewer/refactored/ViewerBottomBar.tsx:149]: Got confused by this because of `getCurrentCanvas`! We really need to rename `canvas` and `currentCanvas` 🙈 this one only cares about the index/number, right?

**Review by rcantin-w — APPROVED** [review]
